src/main/java/be/uantwerpen/minelabs/entity/BohrBlueprintEntity.java

```diff
@@ -176,13 +183,20 @@ public void dropContents() {
         clear();
     }
 
-    public void dropLastItem() {
+    public void dropLastItem(ServerPlayerEntity player) {
```

Comment:
Please move advancement triggering code to a callback rather than changing the interface:

Add function `onPlayerRemovedItem(ItemStack stack, ServerPlayer player, boolean withHook)` that gets called both from attacking the entity and punching the block.

In this way, we can still allow non-player characters to modify the entity if we ever want to do this.